### ISSUE TITLE
Save remote file failed as root privilege

### ISSUE THREAD

**Issue by peiliu0408** [microsoft/vscode-remote-release#8053]

**Comment by bpasero** [comment]
Can you try to reproduce with our nightly insider builds? You can give our preview releases a try from: https://code.visualstudio.com/insiders/

**Comment by peiliu0408** [comment]
> Can you try to reproduce with our nightly insider builds? You can give our preview releases a try from: https://code.visualstudio.com/insiders/

Hi, I tried to reproduce with insider builds, but it seems there is no offline vs-coder-server-Linux-x64.tar.gz for this version (commit id), so whether there is link to download this line?

**Comment by peiliu0408** [comment]
> Can you try to reproduce with our nightly insider builds? You can give our preview releases a try from: https://code.visualstudio.com/insiders/

I found another not elegant method.I also tried another extension. save_as_root, finally it works. 

Still confusing: save_as_root extensions is designed for editing these files with no permissions, but I worked in container, and login as root. Intuitively, there is mo permission problem, and there is nor prompt window show “permission is denied “, only close the whole app, it tells “ Closing the window is taking a bit longer…”

Therefore, How can I locate the real problem in this situation?

**Comment by bpasero** [comment]
What is `save_as_root extensions`?

**Comment by peiliu0408** [comment]
> What is `save_as_root extensions`?

This extension: https://marketplace.visualstudio.com/items?itemName=yy0931.save-as-root

**Comment by bpasero** [comment]
How is that related to the issue, you are using an extension to save as root and that does not work?

**Comment by peiliu0408** [comment]
> How is that related to the issue, you are using an extension to save as root and that does not work?

With this extension, the issue is solved (file could be saved using this extension, but not Ctrl + S). Actually, this tools is designed for edit no permission file, like /etc/ssh/ssh_config, with a user login.

 I log the container as root, in other words, there is no permission problems here. But it still works here, that weird.

**Comment by bpasero** [comment]
Please see if you can find any logged output that could help us find the source of the issue:
* run `code --verbose` from the command line and attach the output
* in Code select **Help | Toggle Developer Tools** and select the **Console**
* attach any output you see in the **Console** you can find

Thanks!

**Comment by peiliu0408** [comment]
As your suggestion, I start the vscode from cmd. Finally, I got something, following the screen shot (only local network between that pc, window 10, and server, Ubuntu 20.04)
![812573F0-F5A3-411C-9C62-64701AECC089](https://user-images.githubusercontent.com/18061950/219613384-6051a233-d05e-4bd2-bc4f-6a879c3b8308.jpeg)

**Comment by bpasero** [comment]
I am not seeing something obvious other than that the save operation never seems to succeed, but why that is is not revealed in those logs.

**Comment by peiliu0408** [comment]
> I am not seeing something obvious other than that the save operation never seems to succeed, but why that is is not revealed in those logs.

Yes. Always tells “found a pending save for versionID2. Seems there is already a save action is token before. This is reason why the first press button CTRL + S on a new created file or new opened file, no a white dot indicator. Actually, the save action is not finished, but spending by some thread I think.

The file system on server is distribution file system. 

Sadly, I still not find what stop the writing action in server system.

**Comment by bpasero** [comment]
Can you try with insiders? You do not have to download a server, we deploy it for you automatically when you connect to the remote.

You can give our preview releases a try from: https://code.visualstudio.com/insiders/

**Comment by peiliu0408** [comment]
> Can you try with insiders? You do not have to download a server, we deploy it for you automatically when you connect to the remote.
> 
> You can give our preview releases a try from: https://code.visualstudio.com/insiders/

Thanks for your kindly support. It seems that the offline vscode-server file could not be download right now,  could you offer a link to download vscode-server.tar.gz file for no Internet connection server?

**Comment by shidahu** [comment]
Hi, @bpasero @peiliu0408 
I encounter the same problem. There is only local network between pc, window 10, and server, Ubuntu 20.04. I can save the newly opened file on the server, but after that, I can't save it again.

**Comment by bpasero** [comment]
Can you try with insiders? You do not have to download a server, we deploy it for you automatically when you connect to the remote.

You can give our preview releases a try from: https://code.visualstudio.com/insiders/

**Comment by peiliu0408** [comment]
Yes, the insider version of VSCode has encountered the same issue. The real problem here is there what caused the saving action failed before figuring out a solution. 
even with verbose mode, only simple messages are showing, but not a Exception with rich info.

**Comment by peiliu0408** [comment]
@shidahu I doubt this diploma is caused by the strong ‘distribution system’,  because the issue occurred in ubuntu 18.04 docker container. Anyway, the extension ‘save_as_root’ works here with uncomfortable using experience.

**Comment by bpasero** [comment]
Can we try the `--verbose` thing again but this time you send the logs over to me in text form? You can do so by opening the logs folder (via command pallette) and zipping the contents up.

Ideally from a session where you only try to save until the issue appears.
